### DISCUSSION TITLE
[Nextcloud] enforce admin configuration on other users

### DISCUSSION THREAD

**Discussion by relikd** [the-djmaze/snappymail#563] [Ideas]
Hi, ok, this request will probably require larger changes and maybe people will say it is unnecessary. But I ask anyway 😝 

My scenario is a Nextcloud instance for a small non-profit team. We use the auto-login feature to automatically assign an email account to specific people. Think of it like collect-all email adresses for compartments. E.g. User (U1) has access to email accounts (E1 and E2), User (U2) has only access to E1 and E3. User (U3) has no access to emails at all. The admin configures the login credentials for all users and the users itself do not know the credentials and have no need to manually configure them. Specifically, if the business case requires, the admin can remove an email account or add new ones.

So, my wish is to pre-configure email logins for users. Basically something like the `override-smtp-credentials` plugin would suffice, except that this plugin can only set a single override (for multiple user) where as I need a m-to-n mapping.

Here are a few shortcoming of the current SnappyMail:
- One account can have multiple sub-accounts (e.g., log in with A but B, C, and D are also loaded). In my case that is actually a plus because I can configure the credentials once and simply log-in with the appropriate account for each user. However, that holds only as long a I have a unique email address to use. For example, I can not have three users with access to A, B, and AB email accounts respectively. If I add email B for the third user, the first user will also gain access to email B. What I currently do is create a fake-account C (with no real use) and add A and B.
- The second issue regards control. A user should not be able to see or change the assigned credentials to prevent accidential account logout. I am thinking of an admin checkbox to disable the `settings/user/additional` settings page.
- Third, I currently use the Impersonate plugin to log-in with another user and configure the credentials. It would be super convenient if the admin could do that from his/her settings page. Ideally I would love to see a list of available email accounts and specify for each email account, who can access it. E.g., E1: U1+U2, E2: U1, E3: U2 (these could be just the Nextcloud usernames)

I do not assume this will be address anytime soon (or ever) but I want to drop it here if anyone in the community wants to develop a plugin for SnappyMail 😃

**Comment by dantefromhell** [comment]
Same use case here.
Especially points 2 & 3 would make admin life soo much easier!

I would assume that it might be helpful to enable admins to preset (default but user changeable)/ hardcode (set by admin, not changeable by user) even more configuration options. Hence I wonder if a general config override mechanism could be possible - hoping this won't pull the conversation off-topic.
